### ISSUE TITLE
Submenus (MenuItem) are not properly aligned

### ISSUE THREAD

**Issue by Xaalek** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2190]
Hello,
As you can see on my screenshot, submenus (MenuItem) are not properly aligned. 
I don’t know if it’s intentional, but I think it's an issue.

![image](https://user-images.githubusercontent.com/54487782/101797198-0a182b80-3b0a-11eb-9231-cacc294e81b5.png)

**Comment by Keboo** [comment]
Agreed this is a bug. Based on the [spec for Cascading menu](https://material.io/components/menus) child list items should have the first item lined up with the parent item.

**Comment by Xaalek** [comment]
@jizc @Keboo 

Ok now the alignement is good when menus open downward

![image](https://user-images.githubusercontent.com/54487782/109933181-6a62d780-7ccb-11eb-897d-3e52659dd950.png)

But we have to take charge of the fact that menus can open upward

![image](https://user-images.githubusercontent.com/54487782/109933124-5c14bb80-7ccb-11eb-91a6-4297cc96a8cb.png)

Putting a negative vertical offset correct the alignement when menus open downward but not upward
Maybe we should check if the menu opens upward or downward before applying offset

**Comment by jizc** [comment]
Ah, I didn't think of this. It makes it a lot more complicated. My best guess is that it might be possible to do with a [CustomPopupPlacementCallback Delegate](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.custompopupplacementcallback?view=net-5.0).
